spec/models/story_spec.rb

```diff
@@ -70,4 +68,24 @@
       end
     end
   end
+
+  describe "#fix_project_start_date" do
+    let(:project)         { create(:project, start_date: nil) }
+    let(:story_params)    { { title: 'Test Story', state: 'started', accepted_at: nil } }
+    let(:story)           { project.stories.build(story_params) }
+
+    it "sets the project start_date to current" do
```

Comment:
You could probably extend this describe block with a few more examples. For instance, it should not set the date to current when:

- The state has changed
- A project does not exist
- A start date already exists
- The current state is not either unstarted or unscheduled

Since you started out with an example that has all the conditions to make it thrive, you could start invalidating one by one individually within new contexts.